src/platforms/linux/linuxnetworkwatcher.cpp

```diff
@@ -0,0 +1,167 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "linuxnetworkwatcher.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "timersingleshot.h"
+
+#include <QtDBus/QtDBus>
+
+// https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html#NMDeviceType
+#ifndef NM_DEVICE_TYPE_WIFI
+#  define NM_DEVICE_TYPE_WIFI 2
+#endif
+
+// https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html#NM80211ApFlags
+#ifndef NM_802_11_AP_SEC_NONE
+#  define NM_802_11_AP_SEC_NONE 0x00000000
+#endif
+
+#ifndef NM_802_11_AP_SEC_PAIR_WEP40
+#  define NM_802_11_AP_SEC_PAIR_WEP40 0x00000001
+#endif
+
+#ifndef NM_802_11_AP_SEC_PAIR_WEP104
+#  define NM_802_11_AP_SEC_PAIR_WEP104 0x00000001
+#endif
+
+namespace {
+Logger logger(LOG_LINUX, "LinuxNetworkWatcher");
+}
+
+static inline bool checkSecurityFlags(int securityFlags) {
+  return securityFlags == NM_802_11_AP_SEC_NONE ||
+         (securityFlags & NM_802_11_AP_SEC_PAIR_WEP40 ||
+          securityFlags & NM_802_11_AP_SEC_PAIR_WEP104);
+}
+
+LinuxNetworkWatcher::LinuxNetworkWatcher(QObject* parent)
+    : NetworkWatcherImpl(parent) {
+  MVPN_COUNT_CTOR(LinuxNetworkWatcher);
+}
+
+LinuxNetworkWatcher::~LinuxNetworkWatcher() {
+  MVPN_COUNT_DTOR(LinuxNetworkWatcher);
+}
+
+void LinuxNetworkWatcher::initialize() {
+  logger.log() << "initialize";
+
+  // Let's wait a few seconds to allow the UI to be fully loaded and shown.
+  // This is not strickly needed, but it's better from a user-experience.
+  TimerSingleShot::create(this, 2000, [this]() {
+    logger.log()
+        << "Retrieving the list of wifi network devices from NetworkManager";
+
+    QDBusInterface nm(
+        "org.freedesktop.NetworkManager", "/org/freedesktop/NetworkManager",
```

Comment:
What about declaring these at the top along with the wifi flags. Keeps all the dbus constants at the same place.